src/librustc/middle/cstore.rs

```diff
@@ -282,10 +282,6 @@ pub fn validate_crate_name(sess: Option<&Session>, s: &str, sp: Option<Span>) {
         if c == '_'  { continue }
         say(&format!("invalid character `{}` in crate name: `{}`", c, s));
```

Comment:
heh, I take it from your changes above that we're better off just dying as soon as we see the first invalid character rather than printing them all?